docs/technical-design/endpoints.md

```diff
@@ -0,0 +1,213 @@
+
+```typescript
+dataExport: () => Promise<DataExportType | undefined>
+```
+
+And then in the NewPostgresStore:
+
+```typescript
+dataExport: () => dataExport(client),
+```
+
+When we access the store in our app, the dataExport method we defined above will be available to us.
+
+### Creating the resolver
+
+We use Apollo Client to interface with our data source and to provide information from the store to the front-end; the methods to do that are called "resolvers".  
+_**Create `/app-api/src/resolvers/dataExport.ts`**_.
+Here's the code.
+
+```typescript
+import { QueryResolvers } from '../gen/gqlServer'
+import { Store } from '../postgres'
+
+export function dataExportResolver(store: Store): QueryResolvers['dataExport'] {
+    return async (_parent, context) => {
+        const name = await store.dataExport()
+        if (name === undefined) {
+            return {
+                name: undefined,
+            }
+        }
+        return name
+    }
+}
+```
+
+The method we just put on the store gets called here. It will return the result of client.state.findFirst() that we defined above. The return type of `QueryResolvers['dataExport']` is a little mysterious. `QueryResolvers` is a generated file that will include the resolver we're defining here. We'll go over how to generate `QueryResolvers` a little further down.
+
+Now we have to add this resolver to our resolver configuration, so that it can be used on the front end.
+
+_**Go to `/app-api/src/resolvers/configureResolvers.ts`**_
+
+We'll import the resolver we just created.
+
+```typescript
+import { dataExportResolver } from './dataExport'
+```
+
+And also add it to the Resolvers object. We've made a query, so this line will be added to the query methods.
+
+```typescript
+dataExport: dataExportResolver(store),
+```
+
+### Generating code
+
+Once we have our backend code in place, we need to run a couple of commands to generate glue types and methods.
+
+_**On the command line, navigate to `/services/app-graphql`.**_  
+_**Run `yarn gqlgen`**_
+
+_**Again on the command line, navigate to `/services/app-api`**_  
+_**Run `yarn prisma generate`**_
+
+### Getting data on the front end
+
+One result of the `generate` commands is to create a hook that we can use on the front end to call our new resolver. The name of the hook will be of the form _useWhatYourResolverIsCalledQuery_ (it will end with "Mutation" if that's what you've created). You can pull it in and get data.
+
+```typescript
+import { useDataExportQuery } from '../../gen/gqlClient'
+const { data, loading, error } = useDataExportQuery()
+```
+
+## Lambda
+
+### Serverless configuration
+
+When we create a handler, we're telling serverless to create a new Lambda function for us in AWS. If we configure it correctly, that function will have access to things like our postgres database, which itself is hosted on AWS. Here's one existing handler configuration, found in `/services/app-api/serverless.yml`.
```

Comment:
```suggestion
Here is an example of serverless configuration for a Lambda that handles an api request, found in `/services/app-api/serverless.yml`.
```

Deleted sections were just moved up. Was trying to match what you did in the GraphQL section - you had an introduction paragraph first.